### PR TITLE
Allow empty provenance actions, just drop them

### PR DESCRIPTION
**Pull request by MrCreosote**
The requirement to have at least one field broke external code

### CODE REVIEW

File: lib/Bio/KBase/workspace/Client.pm [hunk 1]
```diff
@@ -9556,8 +9556,11 @@ A provenance action.
         A provenance action (PA) is an action taken while transforming one data
```
Comment (MrCreosote) [inline lib/Bio/KBase/workspace/Client.pm:9556]: generated, so it Workspace.html above

File: lib/biokbase/workspace/client.py [hunk 2]
```diff
@@ -710,72 +710,72 @@ def save_objects(self, params, context=None):
            action. A provenance action (PA) is an action taken while
```
Comment (MrCreosote) [inline lib/biokbase/workspace/client.py:710]: generated

File: src/us/kbase/workspace/ProvenanceAction.java [hunk 3]
```diff
@@ -20,8 +20,11 @@
  *         A provenance action (PA) is an action taken while transforming one data
```
Comment (MrCreosote) [inline src/us/kbase/workspace/ProvenanceAction.java:20]: generated

File: src/us/kbase/workspace/database/provenance/ProvenanceAction.java [hunk 4]
```diff
@@ -701,12 +701,11 @@ private List<String> processReferences(
...
 		}
 		
-		/** Build the {@link ProvenanceAction}. At least one field must be set before the build
-		 * attempt.
-		 * @return the provenance action.
+		/** Check the builder has any fields set.
+		 * @return true if no fields are set, false otherwise.
 		 */
-		public ProvenanceAction build() {
-			if (	// ew
```
Comment (ialarmedalien) [inline src/us/kbase/workspace/database/provenance/ProvenanceAction.java:709]: hurrah! no more `ew`

**Review by ialarmedalien — APPROVED** [review]
Go forth and multiply! or at least allow provenance to be empty

File: src/us/kbase/workspace/database/provenance/ProvenanceAction.java (hunk 4 above)
Comment (MrCreosote) [reply src/us/kbase/workspace/database/provenance/ProvenanceAction.java:709]: I actually was disappointed when I noticed I removed that. It's still ew in essence, if no longer by name

**Pull request merged**